packages/expo-file-system/android/src/main/java/expo/modules/filesystem/unifiedfile/AssetFile.kt

```diff
@@ -0,0 +1,120 @@
+package expo.modules.filesystem.unifiedfile
+
+import android.content.Context
+import android.net.Uri
+import android.webkit.MimeTypeMap
+import androidx.core.net.toUri
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
+
+class AssetFile(private val context: Context, override val uri: Uri) : UnifiedFileInterface {
+  val path: String = uri.path?.trimStart('/') ?: throw IllegalArgumentException("Invalid asset URI: $uri")
+
+  override fun exists(): Boolean = isDirectory() || isFile()
+
+  override fun isDirectory(): Boolean {
+    return context.assets.list(path)?.isNotEmpty() == true
+  }
+
+  override fun isFile(): Boolean {
+    return try {
+      context.assets.open(path).use {
+        true
+      }
+    } catch (_: Exception) {
+      false
+    }
+  }
+
+  override val parentFile: UnifiedFileInterface?
+    get() {
+      val currentPath = uri.path.orEmpty()
+      if (currentPath.isEmpty()) {
+        return null
+      }
+
+      val parentPath = currentPath.substringBeforeLast('/')
+      val parentUri = "asset://$parentPath".toUri()
+
+      return AssetFile(context, parentUri)
+    }
+
+  override fun createFile(mimeType: String, displayName: String): UnifiedFileInterface? {
+    throw UnsupportedOperationException("Asset files are not writable and cannot be created")
+  }
+
+  override fun createDirectory(displayName: String): UnifiedFileInterface? {
+    throw UnsupportedOperationException("Asset directories are not writable and cannot be created")
+  }
+
+  override fun delete(): Boolean = throw UnsupportedOperationException("Asset files are not writable and cannot be deleted")
+
+  override fun listFilesAsUnified(): List<UnifiedFileInterface> {
+    val list = context.assets.list(path)
+    return list?.map { name -> AssetFile(context, File(path, name).toUri()) as UnifiedFileInterface } ?: emptyList()
+  }
+
+  override val type: String?
+    get() {
+      val extension = MimeTypeMap.getFileExtensionFromUrl(uri.toString())
+      return if (extension.isNotEmpty()) MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension.lowercase()) else null
```

Comment:
not sure where this comes out on the JS side but I just want to say that `getMimeTypeFromExtension` is not 100% reliable. The mime registry is an evolving database https://www.iana.org/assignments/media-types/media-types.xhtml and may not contain "exotic" types. And on iOS, it's probably a similar story, and iOS seems to prefer the use of https://developer.apple.com/documentation/uniformtypeidentifiers/uttype-swift.struct to mime types.

Reply:
Thanks, might be worth it to document in a comment, but Android does not provide any way to access mimetype for assets in any way except by parsing the extension (from my research)